messaging/src/main/java/org/axonframework/eventhandling/deadletter/DeadLetteringCacheEntry.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2010-2023. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.eventhandling.deadletter;
+
+import org.axonframework.eventhandling.EventMessage;
+import org.axonframework.messaging.deadletter.SequencedDeadLetterQueue;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.invoke.MethodHandles;
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.Set;
+
+/**
+ * This will contain a representation of the sequence identifiers belonging to a single segment. This way we know from
+ * memory which identifier is, or is not already present. This in turn reduces the amount of calls directly to a
+ * {@link org.axonframework.messaging.deadletter.SequencedDeadLetterQueue} implementation.
+ *
+ * @author Gerard Klijs
+ * @since 4.9.0
+ */
+class DeadLetteringCacheEntry {
+
+    private static final Logger logger = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+    private final int segmentId;
+    private final boolean startedEmpty;
+    private final int maxSize;
+    private final LinkedList<Object> identifiersNotInDLQ = new LinkedList<>();
```

Comment:
Curious: why use a `LinkedList` for the sequence identifiers?

Reply:
In normal operations, a dead letter queue is completely empty. Worst case, there are a few items in there. Instead of recording the identifiers _not_ in the DLQ, I suggest we watch he items that _are_ in there. With the segment listener approach mentioned earlier, the cache can listen to changes and populate the keys based on segments being claimed or released.

Reply:
The problem with that, is that there is no easy way of knowing which sequence identifiers are already in the dlq, without either making use of implementation details in the dlq, or introducing breaking changes in the dlq api.

Reply:
It's a linked list, so we can easily remove the oldest one when the size becomes too large.